examples/userapi/04_boundary_conditions.ipynb

```diff
@@ -0,0 +1,471 @@
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "density = 1.  # 1000kg/m^3\n",
+    "velocity = 4.  # km/s"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We also need a `TimeFunction` object for each of our wavefields. Note that the pressure field is split in this case: a necessity for implementing the PML boundary conditions. As particle velocity is a vector, we will choose a `VectorTimeFunction` object to encapsulate it."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from devito import TimeFunction, VectorTimeFunction, NODE\n",
+    "\n",
+    "ux = TimeFunction(name='ux', grid=grid, time_order=1,\n",
+    "                  space_order=6, staggered=NODE)\n",
+    "uy = TimeFunction(name='uy', grid=grid, time_order=1,\n",
+    "                  space_order=6, staggered=NODE)\n",
+    "v = VectorTimeFunction(name='v', grid=grid, time_order=1,\n",
+    "                       space_order=6)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A `VectorTimeFunction` is near identical in function to a standard `TimeFunction`, albeit with a field for each grid dimension. The fields associated with each component can be accessed as follows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[[[0. 0. 0. ... 0. 0. 0.]\n",
+      "  [0. 0. 0. ... 0. 0. 0.]\n",
+      "  [0. 0. 0. ... 0. 0. 0.]\n",
+      "  ...\n",
+      "  [0. 0. 0. ... 0. 0. 0.]\n",
+      "  [0. 0. 0. ... 0. 0. 0.]\n",
+      "  [0. 0. 0. ... 0. 0. 0.]]\n",
+      "\n",
+      " [[0. 0. 0. ... 0. 0. 0.]\n",
+      "  [0. 0. 0. ... 0. 0. 0.]\n",
+      "  [0. 0. 0. ... 0. 0. 0.]\n",
+      "  ...\n",
+      "  [0. 0. 0. ... 0. 0. 0.]\n",
+      "  [0. 0. 0. ... 0. 0. 0.]\n",
+      "  [0. 0. 0. ... 0. 0. 0.]]]\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(v[0].data)  # Print the data attribute associated with the x component of v"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You may have also noticed the keyword `staggered` in the arguments when we created these functions. As one might expect, these are used for specifying where derivatives should be evaluated relative to the grid, as required for implementing formulations such as the first-order acoustic wave equation or P-SV elastic. Passing a function `staggered=NODE` specifies that its derivatives should be evaluated at the node. One can also pass `staggered=x` or `staggered=y` to stagger the grid by half a spacing in those respective directions. Additionally, a tuple of dimensions can be passed to stagger in multiple directions (e.g. `staggered=(x, y)`). `VectorTimeFunction` objects have their associated grids staggered by default.\n",
```

Comment:
when you talk about ``staggered``, add a forward link to the staggered grids notebooks?

Reply:
Where are they?